### DISCUSSION TITLE
How to prevent RichEditor from storing an attached image with a default caption and a link to open it directly?

### DISCUSSION THREAD

**Discussion by faridsa** [filamentphp/filament#13515] [Help] [answered]
### Package

Form builder

### Package Version

v3.2.91

### How can we help you?

I'm using RichEditor in a post management form, and I feel that the abbiity to insert images between the text is very useful, but I found that default caption with filename and the link pointing directly to image are two undesired features. 

So my question is: Is there a way to avoid it? 
because I was checking the code inserted in DB and it could be very tricky to clean it from the public views.

Thank you in advance!

**Comment by leandrocfe** [comment]
Maybe via css?

```css
.attachment__caption {
    @apply hidden
}
.attachment__metadata-container{
    @apply hidden
}
```

You need to create a custom theme: https://filamentphp.com/docs/3.x/panels/themes#creating-a-custom-theme

**Reply by faridsa** [reply]
Nice idea Leandro! As far I can see It could works for figcaption but not for the link, because generated html code is as

```
	<figure data-trix-attachment="{&quot;contentType&quot;:&quot;image/jpeg&quot;,&quot;filename&quot;:&quot;nota-03-sec-1.jpg&quot;,&quot;filesize&quot;:240968,&quot;height&quot;:800,&quot;href&quot;:&quot;https://mph.devtest.ar/storage/bSNQp1JzbhGsFIbV5c7vgcCrXVq6bWRVIIH2Oy4e.jpg&quot;,&quot;url&quot;:&quot;https://mph.devtest.ar/storage/bSNQp1JzbhGsFIbV5c7vgcCrXVq6bWRVIIH2Oy4e.jpg&quot;,&quot;width&quot;:1200}" data-trix-content-type="image/jpeg" data-trix-attributes="{&quot;presentation&quot;:&quot;gallery&quot;}" class="attachment attachment--preview attachment--jpg">
		<a href="https://mph.devtest.ar/storage/bSNQp1JzbhGsFIbV5c7vgcCrXVq6bWRVIIH2Oy4e.jpg">
			<img src="https://mph.devtest.ar/storage/bSNQp1JzbhGsFIbV5c7vgcCrXVq6bWRVIIH2Oy4e.jpg" width="1200" height="800">
			<figcaption class="attachment__caption">
				<span class="attachment__name">nota-03-sec-1.jpg</span> <span class="attachment__size">235.32 KB</span>
			</figcaption>
		</a>
	</figure>
```

Therefore, I will need to remove the anchor tag that wraps the img tag. I believe I need to remove it in my controller while retrieving the post and before rendering it,

**Reply by faridsa** [reply, marked as answer]
By the moment I suppress unwanted data in frontend view using `preg_replace` in my controller just like:

```
$post = Post::published->whereSlug($slug)->firstOrFail();
$regex = '/<a[^>]*>(<img[^>]*>).*?<figcaption[^>]*>.*?<\/figcaption>.*?<\/a>/is';
$post->content = preg_replace($regex, '$1', $post->content);
```

**Reply by timomedia** [reply]
> By the moment I suppress unwanted data in frontend view using `preg_replace` in my controller just like:
> 
> ```
> $post = Post::published->whereSlug($slug)->firstOrFail();
> $regex = '/<a[^>]*>(<img[^>]*>).*?<figcaption[^>]*>.*?<\/figcaption>.*?<\/a>/is';
> $post->content = preg_replace($regex, '$1', $post->content);
> ```

Hi @faridsa 
Do you have any other way to do this? Because I am also facing this situation, but I only want it to be deleted if no caption is filled in. If a caption is filled in, it should be displayed, as I need this for SEO. Your method seems to delete all captions in every case, right?

**Comment by horaceho** [comment]
Create a custom CSS file to hide the `figcaption` in `resources/css/my.css`:
```css
figcaption {
    display: none;
}
```
register in `AppServiceProvider` `boot()`, as [in documenttion](https://filamentphp.com/docs/3.x/support/assets#registering-css-files):
```php
namespace App\Providers;

use Filament\Support\Assets\Css;
use Filament\Support\Facades\FilamentAsset;
use Illuminate\Support\ServiceProvider;

class AppServiceProvider extends ServiceProvider
{
    /**
     * Register any application services.
     */
    public function register(): void
    {
        //
    }

    /**
     * Bootstrap any application services.
     */
    public function boot(): void
    {
        FilamentAsset::register([
            Css::make('my-stylesheet', __DIR__ . '/../../resources/css/my.css'),
        ]);
    }
}
```
Pulish the CSS via:
```
php artisan filament:assets
```
